btree/btree.c

```diff
@@ -651,6 +651,480 @@ static int get_tick(struct m0_btree_op *bop)
+		//if intenal node:insert delimiter key to left node
+		if (node_level(level->l_node) > 0) {
+			struct slot slot_for_left_node = {
+			.s_node = level->l_node;
+			.s_idx  = node_count(level->l_node);
+			};
+			node_key(&slot_for_delimiter);
+			slot_for_left_node.s_rec.r_key = slot_for_delimiter.s_rec.r_key;
+			node_set(slot_for_left_node);
+		}
+		node_move(level->l_node, level->l_next, D_RIGHT, NR_MAX,tx);
+		if (oi->i_level[oi->i_used-1].l_idx == node_count(oi->i_level[oi->i_used-1].l_node)) {
+			//i.e l_next belongs to different parent
+			//update delimiter key
+			struct slot slot_for_parent_node = {
+				.s_node = oi->i_level[oi->i_used-1].l_node;
+				.s_idx  = oi->i_level[oi->i_used-1].l_idx-1;
+			};
+			node_key(&slot_for_parent_node);
+			slot_for_delimiter.s_rec.r_key = slot_for_parent_node.s_rec.r_key;
+			node_set(&slot_for_delimiter);
+			//make key at l_idx-1 of parent as null
+		}
+		node_del(oi->i_level[oi->i_used-1].l_node, oi->i_level[oi->i_used-1].l_idx, bop->bo_tx);
+	} else {
+		get_delimiter(oi, D_LEFT, &slot_for_delimiter);
+		if (node_level(level->l_node) > 0) {
+		//if intenal node:insert delimiter key to left node
+			struct slot slot_for_left_node = {
+			.s_node = level->l_prev;
+			.s_idx  = node_count(level->l_prev);
+			};
+			node_key(&slot_for_delimiter);
+			slot_for_left_node.s_rec.r_key = slot_for_delimiter.s_rec.r_key;
+			node_set(slot_for_left_node);
+		}
+		node_move(level->l_prev,level->l_node,D_LEFT, NR_MAX,tx);
+		//update delimiter key
+		struct slot_for_parent_node = {
+			.s_node = oi->i_level[oi->i_used-1].l_node;
+			.s_idx  = oi->i_level[oi->i_used-1].l_idx;
+		};
+		node_key(&slot_for_parent_node);
+		slot_for_delimiter.s_rec.r_key = slot_for_parent_node.s_rec.r_key;
+		node_set(&slot_for_delimiter);
+		node_del(oi->i_level[oi->i_used-1].l_node, oi->i_level[oi->i_used-1].l_idx, bop->bo_tx);
+	}
+}
+/* get_tick() for delete operation : */
+static int get_tick_delete(struct m0_btree_op *bop)
+{
+	struct td             *tree  = (void *)bop->bo_arbor;
+	uint64_t               flags = bop->bo_flags;
+	struct m0_btree_oimpl *oi    = bop->bo_i;
+	struct level          *level = &oi->i_level[oi->i_used];
+
+	switch (bop->bo_op.o_sm.s_state) {
+	case P_INIT:
+		if ((flags & OF_COOKIE) && cookie_is_set(&bop->bo_key.k_cookie))
+			return P_COOKIE;
+		else
+			return P_SETUP;
+	case P_COOKIE:
+		if (cookie_is_valid(tree, &bop->bo_key.k_cookie)
+		&& !possibility_of_UNDERFLOW at cookie_node)
+			return P_LOCK;
+		else
+			return P_SETUP;
+	case P_SETUP:
+		alloc(bop->bo_i, tree->t_height);
+		if (bop->bo_i == NULL)
+			return fail(bop, M0_ERR(-ENOMEM));
+		return P_LOCKALL;
+	case P_LOCKALL:
+		if (bop->bo_flags & OF_LOCKALL)
+			return m0_sm_op_sub(&bop->bo_op, P_LOCK, P_DOWN);
+	case P_DOWN:
+		oi->i_used = 0;
+		/* Load root node. */
+		return node_get(&oi->i_nop, tree, &tree->t_root, P_NEXTDOWN);
+	case P_NEXTDOWN:
+		if (oi->i_nop.no_op.o_sm.sm_rc == 0) {
+			struct slot    slot = {};
```

Comment:
It is bad practice to keep the struct name and its variable's name the same.
Please update it.